File: bongf/level1/mock_test.py
def solution(answers):
    correct = [0, 0, 0]
    first_p = [1, 2, 3, 4, 5]
    second_p = [2, 1, 2, 3, 2, 4, 2, 5]
    third_p = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    answer_list = []

    for i in range(len(answers)):
        if answers[i] == first_p[i%5] :
            correct[0] += 1
        if answers[i] == second_p[i%8] :
            correct[1] += 1
        if answers[i] == third_p[i%10] :
            correct[2] += 1
        
    max_answer = max(correct)
    for i in range(3) :
        if max_answer == correct[i]:
            answer_list.append(i+1)
    
    return answer_list


answers = [1,2,3,4,5]
print(solution(answers))

File: bongf/java/src/level1/MockTest.java
package level1;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class MockTest {
    public static void main(String[] args) {
        int[] answers = {1,2,3,4,5};
        solution(answers);
    }

    public static int[] solution(int[] answers) {
        int[] p1 = {1, 2, 3, 4, 5};
        int[] p2 = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] p3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] correct = {0, 0, 0};

        for( int i=0; i < answers.length; i++) {
            if( answers[i] == p1[i%5] ) {
                correct[0]++;
            }
            if( answers[i] == p2[i%8] ) {
                correct[1]++;
            }
            if( answers[i] == p3[i%10] ) {
                correct[2]++;
            }
        }

        int max = Arrays.stream(correct).max().getAsInt();
        List<Integer> answerL = new ArrayList();
        for ( int i = 0; i < 3; i++ ) {
            if(correct[i] == max ) {
                answerL.add(i+1);
            }
        }
        return answerL.stream().mapToInt(x-> x).toArray();
    }
}
